unity-renderer/Assets/Scripts/MainScripts/DCL/Controllers/LoadingScreenV2/LoadingScreenScripts/LoadingScreenHintsController.cs

```diff
@@ -0,0 +1,135 @@
+using Cysharp.Threading.Tasks;
+using System;
+using System.Collections.Generic;
+using System.Threading;
+using UnityEngine;
+using Object = UnityEngine.Object;
+
+namespace DCL.Controllers.LoadingScreenV2
+{
+    /// <summary>
+    /// Controller responsible of managing the hints views and requesting hints from the HintRequestService
+    /// And also responsible of showing the hints in the LoadingScreen using the HintViewManager carousel
+    /// </summary>
+    public class LoadingScreenHintsController
+    {
+        private readonly TimeSpan SHOWING_TIME_HINTS = TimeSpan.FromSeconds(5);
+        private const int MAX_HINTS = 15;
+
+        private readonly HintView hintViewPrefab;
+        private readonly HintRequestService hintRequestService;
+
+        private bool isRequestingHints = false;
+
+        internal HintViewManager hintViewManager;
+        internal readonly List<HintView> hintViewPool;
+        internal Dictionary<int, Tuple<Hint, Texture2D>> hintsDictionary;
+        internal CancellationTokenSource cancellationTokenSource;
+
+        public event Action OnRequestHintsCompleted;
+
+        public LoadingScreenHintsController(HintView hintViewPrefab, HintRequestService hintRequestService)
+        {
+            this.hintViewPrefab = hintViewPrefab;
+            this.hintRequestService = hintRequestService;
+
+            hintsDictionary = new Dictionary<int, Tuple<Hint, Texture2D>>();
+            hintViewPool = new List<HintView>();
+
+            // Initializing empty hints views
+            for (int i = 0; i < MAX_HINTS; i++)
+            {
+                HintView newHintView = Object.Instantiate(hintViewPrefab);
+                newHintView.ShowHint(false);
+                hintViewPool.Add(newHintView);
+            }
+
+            InitializeHints();
+        }
+
+        private async void InitializeHints()
```

Comment:
I would rename this to `InitializeHintsAsync`, it can also return a UniTaskVoid to avoid using a `.Forget()`